File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
        int i=0,count=0;
        if(ruleKey=="type")
        {
            i=0;
        }
        else if(ruleKey=="color")
        {
            i=1;
        }else
        {
            i=2;
        }
        for(int j=0;j<items.size();j++)
        {
           // cout<<items[j][i]<<"=="<<ruleValue<<"\n";
            if(items[j][i]==ruleValue)
            {
                count++;
            }
        }
        return count;
    }
};

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int i=-1,count=0;
        switch(ruleKey){
            case "type":i=0;
                break;
            case "color":i=1;
                break;
            case "name":i=2;
                break;
        }
        for(int j=0;j<items.size();j++){
            if((items.get(j).get(i)).equals(ruleValue)){
               System.out.println(items.get(j).get(i));
                count++;
            }
        }
        return count;
    }
}